File: support_tools/swigwin-3.0.12/Examples/test-suite/go/overload_complicated_runme.go
package main

import . "./overload_complicated"

func main() {
	var pInt *int

	// Check the correct constructors are available
	p := NewPop(pInt)

	p = NewPop(pInt, false)

	// Check overloaded in const only and pointers/references
	// which target languages cannot disambiguate
	if p.Hip(false) != 701 {
		panic("Test 1 failed")
	}

	if p.Hip(pInt) != 702 {
		panic("Test 2 failed")
	}

	// Reverse the order for the above
	if p.Hop(pInt) != 805 {
		panic("Test 3 failed")
	}

	if p.Hop(false) != 801 {
		panic("Test 4 failed")
	}

	// Few more variations and order shuffled
	if p.Pop(false) != 901 {
		panic("Test 5 failed")
	}

	if p.Pop(pInt) != 902 {
		panic("Test 6 failed")
	}

	if p.Pop() != 905 {
		panic("Test 7 failed")
	}

	// Overload on const only
	if p.Bop(pInt) != 1001 {
		panic("Test 8 failed")
	}

	if p.Bip(pInt) != 2001 {
		panic("Test 9 failed")
	}

	// Globals
	if Muzak(false) != 3001 {
		panic("Test 10 failed")
	}

	if Muzak(pInt) != 3002 {
		panic("Test 11 failed")
	}
}

File: support_tools/swigwin-3.0.12/Examples/test-suite/octave/overload_complicated_runme.m
# do not dump Octave core
if exist("crash_dumps_octave_core", "builtin")
  crash_dumps_octave_core(0);
endif

overload_complicated

pInt = None;

# Check the correct constructors are available
p = Pop(pInt);

p = Pop(pInt, 0);

# Check overloaded in const only and pointers/references which target languages cannot disambiguate
if (p.hip(0) != 701)
  error("Test 1 failed")
endif

if (p.hip(pInt) != 702)
  error("Test 2 failed")
endif

# Reverse the order for the above
if (p.hop(pInt) != 805)
  error("Test 3 failed")
endif

if (p.hop(0) != 801)
  error("Test 4 failed")
endif

# Few more variations and order shuffled
if (p.pop(0) != 901)
  error("Test 5 failed")
endif

if (p.pop(pInt) != 902)
  error("Test 6 failed")
endif

if (p.pop() != 905)
  error("Test 7 failed")
endif

# Overload on const only
if (p.bop(pInt) != 1001)
  error("Test 8 failed")
endif

if (p.bip(pInt) != 2001)
  error("Test 9 failed")
endif

# Globals
if (muzak(0) != 3001)
  error("Test 10 failed")
endif

if (muzak(pInt) != 3002)
  error("Test 11 failed")
endif

